sdk/storage/Azure.Storage.Files/tests/FileTestBase.cs

```diff
@@ -0,0 +1,201 @@
+                TestConfigurations.DefaultTargetTenant.AccountKey);
+
+        public SasQueryParameters GetNewAccountSasCredentials(SharedKeyCredentials sharedKeyCredentials = default)
+            => new AccountSasSignatureValues
+            {
+                Protocol = SasProtocol.None,
+                Services = new AccountSasServices { File = true }.ToString(),
+                ResourceTypes = new AccountSasResourceTypes { Container = true }.ToString(),
+                StartTime = this.Recording.UtcNow.AddHours(-1),
+                ExpiryTime = this.Recording.UtcNow.AddHours(+1),
+                Permissions = new FileAccountSasPermissions { Create = true, Delete = true }.ToString(),
+                IPRange = new IPRange { Start = IPAddress.None, End = IPAddress.None }
+            }.ToSasQueryParameters(sharedKeyCredentials);
+
+        public SasQueryParameters GetNewFileServiceSasCredentialsShare(string shareName, SharedKeyCredentials sharedKeyCredentials = default)
+            => new FileSasBuilder
+            {
+                ShareName = shareName,
+                Protocol = SasProtocol.None,
+                StartTime = this.Recording.UtcNow.AddHours(-1),
+                ExpiryTime = this.Recording.UtcNow.AddHours(+1),
+                Permissions = new ShareSasPermissions { Read = true, Write = true, List = true, Create = true, Delete = true }.ToString(),
+                IPRange = new IPRange { Start = IPAddress.None, End = IPAddress.None }
+            }.ToSasQueryParameters(sharedKeyCredentials ?? this.GetNewSharedKeyCredentials());
+
+        public SasQueryParameters GetNewFileServiceSasCredentialsFile(string shareName, string filePath, SharedKeyCredentials sharedKeyCredentials = default)
+            => new FileSasBuilder
+            {
+                ShareName = shareName,
+                FilePath = filePath,
+                Protocol = SasProtocol.None,
+                StartTime = this.Recording.UtcNow.AddHours(-1),
+                ExpiryTime = this.Recording.UtcNow.AddHours(+1),
+                Permissions = new FileSasPermissions { Read = true, Write = true, Create = true, Delete = true }.ToString(),
+                IPRange = new IPRange { Start = IPAddress.None, End = IPAddress.None }
+            }.ToSasQueryParameters(sharedKeyCredentials ?? this.GetNewSharedKeyCredentials());
+
+        public SignedIdentifier[] BuildSignedIdentifiers() =>
+            new[]
+            {
+                new SignedIdentifier
+                {
+                    Id = this.GetNewString(),
+                    AccessPolicy =
+                        new AccessPolicy
+                        {
+                            Start =  this.Recording.UtcNow.AddHours(-1),
+                            Expiry =  this.Recording.UtcNow.AddHours(1),
+                            Permission = "rw"
+                        }
+                }
+            };
+
+        class DisposingShare : IDisposable
```

Comment:
Should we consider using `IAsyncDisposable` instead of `Wait()`?

Reply:
I'm more comfortable with this right now because I know any exceptions will be correctly swallowed.  I'll added #6635 to track this for the next milestone.